.github/workflows/validate-json.yml

```diff
@@ -0,0 +1,42 @@
+name: Validate JSON
+on:
+  workflow_call:
+    inputs:
+      src:
+        type: string
+        required: false
+        default: "."
+        description: "A directory that contains both the *.json/*.schema.json files (defaults to '.')"
+
+     
+jobs:
+  setup-validate-json:
+    name: Setup Validate JSON
+    runs-on: ubuntu-latest
+    outputs:
+      json-to-validate: ${{ steps.get-json.outputs.json }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get all schema 
+        id: get-json
+        # this command finds all the filenames with the '.schema.json' extension, removes that extension, then turns them into a json array
+        # an example of this transformation would be: 'compilers.schema.json models.schema.json' -> 'compilers models' -> '["compilers","models"]' 
+        run: echo "json=$(find ${{ inputs.src }} -type f -name '*.schema.json' | xargs basename --suffix '.schema.json' | jq --raw-input --null-input --compact-output '[inputs]')" >> $GITHUB_OUTPUT
+
+  validate-json:
+    name: Validate JSON
+    runs-on: ubuntu-latest
+    needs:
+      - setup-validate-json
+    strategy:
+      fail-fast: false
+      matrix: 
+        file: ${{ fromJson(needs.setup-validate-json.outputs.json-to-validate) }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
+        with:
+          python-version: '3.10'
+      - name: Validate files
+        run: jsonschema --instance ${{ inputs.src }}/${{ matrix.file }}.json ${{ inputs.src }}/${{ matrix.file }}.schema.json
```

Comment:
Comment to future self (and anyone else listening in), `jsonschema` is from the [python-jsonschema package](https://github.com/python-jsonschema/jsonschema) which is installed with `pip` in the `actions/setup-python@v3` workflow.

The code for the CLI that is being used is here:

https://github.com/python-jsonschema/jsonschema/blob/main/jsonschema/cli.py